plugins/CoreVisualizations/Visualizations/JqplotGraph/Evolution.php

```diff
@@ -51,42 +53,44 @@ public function beforeLoadDataTable()
 
         parent::beforeLoadDataTable();
 
-        // period will be overridden when 'range' is requested in the UI
-        // but the graph will display for each day of the range.
-        // Default 'range' behavior is to return the 'sum' for the range
-        if (Common::getRequestVar('period', false) == 'range') {
-            $this->requestConfig->request_parameters_to_modify['period'] = 'day';
+        // period will be overridden when 'range' is requested in the UI.
+        // The graph will display the range in the most suitable period and
+        // it won't show historical data before the range.
+        $period = Common::getRequestVar('period', false);
+        $selector = StaticContainer::get(EvolutionPeriodSelector::class);
+
+        if ($period === 'range') {
+            $date = Common::getRequestVar('date', false);
+            $requestingPeriod = Factory::build($period, $date);
+
+            // if a larger date range is selected, then for better performance and for seeing trends better we want to use
+            // a suitable period (rather than always using for example the day range)
+            $this->requestConfig->request_parameters_to_modify['period'] = $selector->getHighestPeriodInCommon($requestingPeriod, []);
```

Comment:
When now viewing a large range, the evolution graph may switch from day period to week or month or year to needing to fetch less archives for faster performance and it allows detecting trends better